include/mscclpp/semaphore_device.hpp

```diff
@@ -20,14 +20,22 @@ struct Host2DeviceSemaphoreDeviceHandle {
   /// @return true if the host has signaled.
   MSCCLPP_DEVICE_INLINE bool poll() {
     bool signaled = (atomicLoad(inboundSemaphoreId, memoryOrderAcquire) > (*expectedInboundSemaphoreId));
-    if (signaled) (*expectedInboundSemaphoreId) += 1;
+    if (signaled) {
+      (*expectedInboundSemaphoreId) += 1;
+    } else {
+      // TODO: MSRCHA-371
+      atomicStore(&inboundSemaphoreId[1], uint64_t{0}, memoryOrderRelaxed);
+    }
```

Comment:
Seems we can remove this and following workaround for MSRCHA-371

Reply:
Thanks, reverted this change.